packages/theme/src/cli/services/check.ts

```diff
@@ -0,0 +1,359 @@
+import {fileExists, readFileSync, writeFile} from '@shopify/cli-kit/node/fs'
+import {outputInfo, outputSuccess} from '@shopify/cli-kit/node/output'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {renderInfo, renderTasks, type Task} from '@shopify/cli-kit/node/ui'
+import {
+  Severity,
+  applyFixToString,
+  autofix,
+  loadConfig,
+  themeCheckRun,
+  type FixApplicator,
+  type Offense,
+  type Theme,
+  type ThemeCheckRun,
+} from '@shopify/theme-check-node'
+import YAML from 'yaml'
+
+interface OffenseMap {
+  [check: string]: Offense[]
+}
+
+interface TransformedOffense {
+  check: string
+  severity: number
+  start_row: number
+  start_column: number
+  end_row: number
+  end_column: number
+  message: string
+}
+
+interface TransformedOffenseMap {
+  path: string
+  offenses: TransformedOffense[]
+  errorCount: number
+  suggestionCount: number
+  styleCount: number
+}
+export type FailLevel = 'error' | 'suggestion' | 'style'
+
+function severityToFailLevel(severity: Severity): FailLevel {
+  switch (severity) {
+    case Severity.ERROR:
+      return 'error'
+    case Severity.WARNING:
+      return 'suggestion'
```

Comment:
More of a pet peeve than anything. If we're to render it, we should render it with the "modern" names. We support those for backwards compat, but we should print them as the new ones (`info`, `warning`, and`error`) See https://github.com/Shopify/theme-tools/blob/main/packages/theme-check-node/src/config/types.ts#L47-L56